munit/shared/src/main/scala/munit/Assertions.scala

```diff
@@ -67,20 +54,25 @@ trait Assertions extends MacroCompat.CompileErrorMacro {
       Diffs.assertNoDiff(
         obtained,
         expected,
-        munitComparisonHandler(obtained, expected),
+        ComparisonFailExceptionHandler.fromAssertions(this, Clues.empty),
         munitPrint(clue),
         printObtainedAsStripMargin = true
       )
     }
   }
 
+  /**
+   * Asserts that two elements are not equal according to the `Compare[A, B]` type-class.
+   *
+   * By default, uses `==` to compare values.
+   */
   def assertNotEquals[A, B](
       obtained: A,
       expected: B,
       clue: => Any = "values are the same"
-  )(implicit loc: Location, ev: A =:= B): Unit = {
+  )(implicit loc: Location, ev: Compare[A, B]): Unit = {
```

Comment:
```suggestion
  )(implicit loc: Location, compare: Compare[A, B]): Unit = {
```

for consistency with `assertEquals`

Reply:
👍 Done